File: dashboard/src/types/category.ts
// Category types matching the real backend API (Spanish field names)

export type CategoryEstado = 'activo' | 'inactivo';

export interface Category {
  id: number;
  nombre: string;
  icono: string | null;
  imagen_url: string | null;
  orden: number;
  estado: CategoryEstado;
  es_home: boolean;
  created_at: string;
  updated_at: string;
}

export interface CategoryCreate {
  nombre: string;
  icono?: string | null;
  imagen_url?: string | null;
  orden?: number;
  estado?: CategoryEstado;
  es_home?: boolean;
}

export interface CategoryUpdate {
  nombre?: string;
  icono?: string | null;
  imagen_url?: string | null;
  orden?: number;
  estado?: CategoryEstado;
  es_home?: boolean;
}

File: rest_api/app/schemas/category.py
"""Category schemas — Spanish API field names."""

from __future__ import annotations

from pydantic import BaseModel, ConfigDict, Field
from typing import Annotated


class CategoryRead(BaseModel):
    """Category response shape matching spec section 3.4."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    nombre: str
    icono: str | None = None
    imagen_url: str | None = None
    orden: int = 0
    estado: str  # "activo" | "inactivo"
    es_home: bool = False
    created_at: str
    updated_at: str


class CategoryCreate(BaseModel):
    """POST /api/v1/branches/{branch_id}/categories request body."""

    nombre: Annotated[str, Field(min_length=2, max_length=100)]
    icono: str | None = None
    imagen_url: str | None = None
    orden: int | None = None
    estado: str = "activo"


class CategoryUpdate(BaseModel):
    """PUT request body — all fields optional."""

    nombre: Annotated[str | None, Field(min_length=2, max_length=100)] = None
    icono: str | None = None
    imagen_url: str | None = None
    orden: int | None = None
    estado: str | None = None


class CategoryDeleteResponse(BaseModel):
    """Response for category deletion with cascade counts."""

    message: str
    cascade: dict[str, int]
